appstudio-controller/controllers/appstudio.redhat.com/environment_controller.go

```diff
@@ -541,3 +607,90 @@ func (r *EnvironmentReconciler) findObjectsForDeploymentTarget(dt client.Object)
 
 	return envRequests
 }
+
+// findObjectsForSecret finds all the Environment objects that are using this incoming secret.
+// There are two types of secrets that we want to reconcile:
+// 1. Secret created by the SpaceRequest controller
+// 2. Secret created for the managed Environment
+func (r *EnvironmentReconciler) findObjectsForSecret(secret client.Object) []reconcile.Request {
+	ctx := context.Background()
+	handlerLog := log.FromContext(ctx).
+		WithName(logutil.LogLogger_managed_gitops)
+
+	secretObj, ok := secret.(*corev1.Secret)
+	if !ok {
+		handlerLog.Error(nil, "incompatible object in the Environment mapping function, expected a Secret")
+		return []reconcile.Request{}
+	}
+
+	// Check if the secret is created by the Environment controller
+	if secretObj.Type == sharedutil.ManagedEnvironmentSecretType {
+		envName := secretObj.GetLabels()[managedEnvironmentSecretLabel]
+		if envName != "" {
+			return []reconcile.Request{
+				{
+					NamespacedName: types.NamespacedName{
+						Name:      envName,
+						Namespace: secret.GetNamespace(),
+					},
+				},
+			}
+		}
+		return []reconcile.Request{}
+	}
+
+	// If the secret is created by the SpaceRequest controller, find the corresponding Environment.
```

Comment:
Looking at what we're doing here, I'm slightly worried about the amount of work our Environment controller has to do.

AFAICT this code will get called for any Secret that is created/modified on the cluster, rather than just Secrets that are related to our (or space request) API (and RHTAP Prod has ~2500 Secrets).  And for each Secret event we receive, the code we run looks pretty heavy: 2 (namespace-scoped) lists and 1 or more Gets (though hopefully the controller-runtime cache helps reduce this).

In order to significantly reduce this issue, I think we can filter out Secrets that have any of these types:
```
kubernetes.io/dockerconfigjson
kubernetes.io/service-account-token
kubernetes.io/dockercfg
kubernetes.io/tls
```

That reduces the number of Secrets that our code will process event for from ~2500 to (currently) about ~350.

WDYT?

Reply:
Good idea. I've added a check to filter the secrets of other types.